document/src/main/java/com/ritense/document/service/impl/JsonSchemaDocumentDefinitionService.java

```diff
@@ -159,6 +159,25 @@ public Optional<JsonSchemaDocumentDefinition> findLatestByName(String documentDe
         return Optional.ofNullable(definition);
     }
 
+    @Override
+    public Optional<JsonSchemaDocumentDefinition> findByNameAndVersion(String documentDefinitionName, long version) {
+        final DocumentDefinition.Id documentDefinitionId = JsonSchemaDocumentDefinitionId.existingId(documentDefinitionName, version);
+        final var definition = documentDefinitionRepository.findById(documentDefinitionId).orElse(null);
```

Comment:
This could be a map. You're converting from an optional to a nullable variable, then do a nullcheck and then making ann optional again. The map method on Optional are meant to apply a tranformation if the value exists.